### ISSUE TITLE
Generating to new unlabeled data

### ISSUE THREAD

**Issue by selveszero** [wjko2/Domain-Agnostic-Sentence-Specificity-Prediction#8]
Looking at the instructions it seems like I should only make these following changes when trying to get the specificity on new unlabeled data:

In train.py:
if params.test_data=="twitter":
    _,xst= getFeatures(os.path.join(params.nlipath,'twitters.txt'))
    _,xsu = getFeatures('[PATH_TO_MY_UNLABELED_DATA]')

In test.py:
if params.test_data=="twitter":
    _,xst= getFeatures('[PATH_TO_MY_TEST_DATA]')
    _,xsu = getFeatures('[PATH_TO_MY_UNLABELED_DATA]')

In data2.py:
    if dat=='twitter':
        s1['test']['path'] = '[PATH_TO_MY_TEST_DATA]'
        target['test']['path'] = os.path.join(data_path,'twitterl.txt')
        targetv['test']['path'] = 'dataset/data/twitterv.txt'
        s1['unlab']['path'] ='[PATH_TO_MY_UNLABELED_DATA]'

However, does it really not matter that the other variables still use twitter data?

Also, is it ok that my test data is the same as unlabeled data?

**Comment by wjko2** [comment]
It does not matter that the other variables still use twitter data, those files are not used in training result. 
Test data can be same as unlabeled data.

**Comment by JoshuaMathias** [comment]
@wjko2  It doesn't make sense that the target files are not used during training. First of all, if they're missing there's an error. And in the training code a torch variable is created with target in the trainepoch function. If both the train script and the test script are used for fine-tuning on new data, what is used for the initial supervised part of the training, or does this happen as part of train? The distinction between supervised data training and unsupervised domain adaptation is not clarified in the README.

My hypothesis is that whatever the target paths are set to is what is going to be used for the supervised portion of the training. Is this correct? But if this is so, the corresponding input text files would be needed. So how or when is the supervised training occuring?

This is important to understand so I can control what data is used for both the supervised training and the unsupervised training.

Thanks!
PS I also created a fork that doesn't require altering the code for using new data and switching between GPU and CPU: https://github.com/JoshuaMathias/Domain-Agnostic-Sentence-Specificity-Prediction

**Comment by wjko2** [comment]
target['test']['path'] and targetv['test']['path']  are dummy variables, but if those dummy variables are missing there will still be an error. 

The train.py script performs both supervised and unsupervised portion of the training, the labeled training data for supervision should always be 'data.txt', specified in s1['train']['path'] and xsl, and unlabeled data on the target domain are specified in s1['unlab']['path'] and xsu.
In test.py , the script reads target domain test data from s1['test']['path'] and xst.

**Comment by JoshuaMathias** [comment]
@wjko2  Thank you for the quick answer! I also found a corresponding label.txt with the same number of lines, and I see this is described by section 4.1 of the paper.
